lib/ModuleFilenameHelpers.js

```diff
@@ -160,3 +165,7 @@ ModuleFilenameHelpers.matchObject = function matchObject(obj, str) {
 		if(ModuleFilenameHelpers.matchPart(str, obj.exclude)) return false;
 	return true;
 };
+
+ModuleFilenameHelpers.setModuleNamespace = function setModuleNamespace(moduleNamespace) {
+	this._moduleNamespace = moduleNamespace;
```

Comment:
Don't use static variables.

Instead pass the `namespace` to the `createFilename` method.

`createFilename(module, options, requestShortener)`

`options: { namespace, moduleFilenameTemplate }`

Also add a backward-compatibility layer when `typeof options === "string"`